airflow/container.go

```diff
@@ -39,6 +39,7 @@ type RegistryHandler interface {
 // ImageHandler defines methods require to handle all operations on/for container images
 type ImageHandler interface {
 	Build(config types.ImageBuildConfig) error
+	BuildWithoutDags(config types.ImageBuildConfig) error
```

Comment:
I think it should be the responsibility of the caller function and not ImageHandler to take care of adding the `dags` folder to dockerignore before building the image. ImageHandler should have logic just related to doing an operation with an image.

In an ideal world, this is how I had imagined the `ImageHandler` & `ContainerHandler` interfaces to be a set of methods that only deals with a particular Container engine for a specific operation related to Image/Container and wouldn't be aware of anything related to airflow logic or in this case `astro deploy`. 

The reasoning behind this was to make sure that we are able to add more container engines like Podman fairly straightforwardly by simply implementing these methods without having to worry about any airflow logic. I think we have already introduced a couple of methods that shouldn't be here and I might have a go at cleaning those up when working on the Podman feature.

Reply:
Yeah, that makes sense...let me move that logic to the caller function and update tests